src/installer/managed/Microsoft.NET.HostModel/MachO/CodeSignature.cs

```diff
@@ -0,0 +1,155 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.IO.MemoryMappedFiles;
+using System.Linq;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.NET.HostModel.MachO;
+
+internal class CodeSignature
+{
+    private (EmbeddedSignatureHeader Header, long FileOffset) _embeddedSignature;
+    private (CodeDirectoryHeader Header, long FileOffset) _codeDirectory;
+    private (byte[] Identifier, long FileOffset) _identifierPtr;
+    private (byte[] Blob, long FileOffset) _cdHashes;
+    private (RequirementsBlob Header, long FileOffset) _requirementsBlob;
```

Comment:
I can't seem to find where this is actually required for code signing. That is, shouldn't the requirements already be present in the binary?

Reply:
They're present in signed binaries, but not in unsigned ones, and the offset will change if the identifier used is different than the existing signature.

Reply:
But this is only for signing our own binaries, right? So we can assume that the incoming binaries haven't been signed?